pkg/recommendation/recommender/resource/recommend.go

```diff
@@ -0,0 +1,180 @@
+package resource
+
+import (
+	"encoding/json"
+	"fmt"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
+
+	predictionapi "github.com/gocrane/api/prediction/v1alpha1"
+
+	"github.com/gocrane/crane/pkg/metricnaming"
+	"github.com/gocrane/crane/pkg/prediction/config"
+	"github.com/gocrane/crane/pkg/recommend/types"
+	"github.com/gocrane/crane/pkg/recommendation/framework"
+	"github.com/gocrane/crane/pkg/utils"
+)
+
+const callerFormat = "RecommendationCaller-%s-%s"
+
+func (rr *ResourceRecommender) PreRecommend(ctx *framework.RecommendationContext) error {
+	return nil
```

Comment:
no need load algoritjm config?

Reply:
every container has two algoritjm config: cpu and memory ,so cannot use context's algoritjm config. we may need to reconsider it.